crates/core_simd/src/masks/full_masks.rs

```diff
@@ -0,0 +1,301 @@
+//! Masks that take up full SIMD vector registers.
+
+/// The error type returned when converting an integer to a mask fails.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct TryFromMaskError(());
+
+impl core::fmt::Display for TryFromMaskError {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        write!(
+            f,
+            "mask vector must have all bits set or unset in each lane"
+        )
+    }
+}
+
+macro_rules! define_mask {
+    { $(#[$attr:meta])* struct $name:ident<const $lanes:ident: usize>($type:ty); } => {
+        $(#[$attr])*
+        #[derive(Copy, Clone, Default, PartialEq, PartialOrd, Eq, Ord, Hash)]
+        #[repr(transparent)]
+        pub struct $name<const $lanes: usize>($type);
+
+        impl<const $lanes: usize> $name<$lanes> {
+            /// Construct a mask by setting all lanes to the given value.
+            pub fn splat(value: bool) -> Self {
+                Self(<$type>::splat(
+                    if value {
+                        -1
+                    } else {
+                        0
+                    }
+                ))
```

Comment:
I did some poking with the version with the u64 mask and it seems to usually optimize to be branchless up there, but I wonder if this version does as consistently? Could be
```suggestion
                Self(<$type>::splat(value as $type).neg())
```
if it doesn't? I don't know for sure.

Reply:
Yeah, I suppose this is more explicitly branchless.

Reply:
Ah nevermind, I remember why I wrote it this way, I just didn't want to pass in the scalar type as well.  You can't do `value as $type`, `$type` is the vector type.